gpu/gpu.go

```diff
@@ -173,6 +175,34 @@ func GetCPUInfo() GpuInfoList {
 	return GpuInfoList{cpus[0].GpuInfo}
 }
 
+func DetectInteliGpuMemStatus(gpuInfo *OneapiGPUInfo) {
+	var totalram uint64 = uint64(C.check_total_host_mem())
+	// there will be half of total ram can be handle as iGPU vram
+	gpuInfo.TotalMemory = totalram / 2
+	cmd := "sed 's/^.*, \\([0-9]\\+\\) bytes$/\\1/' /sys/kernel/debug/dri/0/i915_gem_objects | awk '{print $1; exit}'"
+	terminal := "bash"
+	if runtime.GOOS == "windows" {
+		cmd = `(((Get-Counter "\GPU Process Memory(*)\Local Usage").CounterSamples | where CookedValue).CookedValue | measure -sum).sum`
+		terminal = "powershell"
+	}
+	output, err := exec.Command(terminal, "-c", cmd).Output()
+
+	if err != nil {
+		fmt.Println("Error executing get Intel iGPUs vram command:", err)
+		return
+	}
+	result := strings.TrimSpace(string(output))
+	if result != "" {
+		usedVRAM, _ := strconv.ParseUint(result, 10, 64)
+		gpuInfo.FreeMemory = gpuInfo.TotalMemory - usedVRAM
+	} else {
+		gpuInfo.GpuInfo.UnreliableFreeMemory = true
+		slog.Warn("can't get Intel iGPU allocated VRAM so there may exist OOM risk during inference, please try to run ollama with sudo privilege. turn UnreliableFreeMemory flag On")
+		gpuInfo.FreeMemory = gpuInfo.TotalMemory - 1024*1024*1024 //leave 1G VRAM for graphic-related task besides ollama.
```

Comment:
IMHO it feels odd referring to normal system RAM as video RAM, although it's just temporarily (MMU) assigned to GPU, not HW that's local to GPU.

Maybe:
```suggestion
		fmt.Println("Error executing command for getting Intel iGPUs system RAM usage:", err)
		return
	}
	result := strings.TrimSpace(string(output))
	if result != "" {
		usedRAM, _ := strconv.ParseUint(result, 10, 64)
		gpuInfo.FreeMemory = gpuInfo.TotalMemory - usedRAM
	} else {
		gpuInfo.GpuInfo.UnreliableFreeMemory = true
		slog.Warn("failed to get amount of system RAM allocated to Intel iGPU, so there exists OOM risk during inference.  As debugfs can be accessed only as root,  please try to run ollama with sudo privilege. Turn UnreliableFreeMemory flag On")
		gpuInfo.FreeMemory = gpuInfo.TotalMemory - 1024*1024*1024 //leave 1G system RAM for other (CPU and iGPU) tasks.
```

Reply:
make sense